src/System.Text.Utf8/src/System/Text/Utf8/Utf8String.cs

```diff
@@ -175,6 +188,11 @@ public override string ToString()
             return new string(characters);
         }
 
+        public bool ReferenceEquals(Utf8String other)
```

Comment:
I can imagine this being an interesting feature in other scenarios as well. Since it diverges from the classic Reference Equals mechanism, I'd like to get some opinions :)